shimmingtoolbox/masking/mask_utils.py

```diff
@@ -0,0 +1,202 @@
+    """
+    Select the appropriate slices from ``nii_mask_from`` using ``from_slices`` and resample onto ``nii_target``
+
+    Args:
+        nii_mask_from (nib.Nifti1Image): Mask to resample from. False or 0 signifies not included.
+        nii_target (nib.Nifti1Image): Target image to resample onto.
+        from_slices (tuple): Tuple containing the slices to select from nii_mask_from.
+        dilate (int): Length of a side of the 3d kernel to dilate the mask. Must be odd.
+
+    Returns:
+        nib.Nifti1Image: Mask resampled with nii_target.shape and nii_target.affine.
+    """
+
+    mask_from = nii_mask_from.get_fdata()
+
+    # Initialize a sliced mask and select the slices from from_slices
+    sliced_mask = np.full_like(mask_from, fill_value=False)
+    sliced_mask[:, :, from_slices] = mask_from[:, :, from_slices]
+
+    # Create nibabel object
+    nii_mask = nib.Nifti1Image(sliced_mask.astype(int), nii_mask_from.affine, header=nii_mask_from.header)
+
+    # Resample the mask onto nii_target
+    nii_mask_target = resample_from_to(nii_mask, nii_target, order=0, mode='grid-constant', cval=0)
+
+    # dilate the mask to add more pixels in particular directions
+    mask_dilated = dilate_binary_mask(nii_mask_target.get_fdata(), 'line', dilate)
+    nii_mask_dilated = nib.Nifti1Image(mask_dilated, nii_mask_target.affine, header=nii_mask_target.header)
+
+    # #######
+    # # Debug
+    # import os
+    # nib.save(nii_mask, os.path.join(os.curdir, f"fig_mask_{from_slices[0]}.nii.gz"))
+    # nib.save(nii_mask_from, os.path.join(os.curdir, "fig_mask_roi.nii.gz"))
+    # nib.save(nii_mask_target, os.path.join(os.curdir, f"fig_mask_res{from_slices[0]}.nii.gz"))
+    # nib.save(nii_mask_dilated, os.path.join(os.curdir, f"fig_mask_dilated{from_slices[0]}.nii.gz"))
+    # #######
+
+    return nii_mask_dilated
+
+
+def dilate_binary_mask(mask, shape='sphere', size=3):
+    """
+    Dilates a binary mask according to different shapes and kernel size
+
+    Args:
+        mask (numpy.ndarray): 3d array containing the binary mask.
+        shape (str): 3d kernel to perform the dilation. Allowed shapes are: 'sphere', 'cross', 'line', 'cube'.
+                     'line' uses 3 line kernels to extend in each directions by "(size - 1) / 2" only if that direction
+                     is smaller than (size - 1) / 2
+        size (int): Length of a side of the 3d kernel. Must be odd.
+
+    Returns:
+        numpy.ndarray: Dilated mask.
+
+    Notes:
+
+        Kernels for
+
+            * 'cross':
+                ::
+
+                      np.array([[[0, 0, 0],
+                                 [0, 1, 0],
+                                 [0, 0, 0]],
+                                [[0, 1, 0],
+                                 [1, 1, 1],
+                                 [0, 1, 0]],
+                                [[0, 0, 0],
+                                 [0, 1, 0],
+                                 [0, 0, 0]]])
```

Comment:
Is `cross[2]` the same as `cross[0]` or is it a mistake ? Also the size of the kernel is only specified for `sphere`.

Reply:
Its a 3x3 kernel where cross[0] == cross[2]. I added a size comment for the other ones.